src/language/branches.md

```diff
@@ -0,0 +1,295 @@
+---
+title: Branches 
+description: Learn how to use branches to control the flow of your Dart code.
+---
+
+You can control the flow of your Dart code using branches:
+
+-   `if`, `case`, and `else`
+-   `switch` and `case` statements
+-   `switch` and `case` expressions
+
+Other methods of manipulating control flow in Dart are:
+
+-   [`for` loops][]
+-   [`while` and `do while` loops][]
+-   [`break` and `continue`][break]
+-   [`assert`][]
+
+You can also affect control flow using [Exceptions][], like `try-catch` and `throw`.
+
+## If else
+
+Dart supports `if` statements with optional `else` statements:
+
+<?code-excerpt "misc/lib/language_tour/control_flow.dart (if-else)"?>
+```dart
+if (isRaining()) {
+  you.bringRainCoat();
+} else if (isSnowing()) {
+  you.wearJacket();
+} else {
+  car.putTopDown();
+}
+```
+
+The statement conditions must be expressions
+that evaluate to [boolean values][boolean], nothing else.
+To learn how to use `if` in an expression context, see [Conditional expressions][]
+
+## If case
+
+Dart `if` statements support `case` clauses followed by a [case pattern][pattern]: 
```

Comment:
I don't think there's a need to introduce a new concept of 'case patterns'. The patterns occurring in this syntactic context are just patterns.

Of course, we do have the notions of 'declaration context', 'matching context', 'assignment context', and you could argue that we have different kinds of patterns in those contexts. (For instance, an identifier pattern has a very different meaning in each of those contexts.) But it's quite reasonable to say that we just have patterns, and they can occur in each of those different contexts. With that, we don't need to introduce different words to denote patterns when they occur in different contexts. I think that's going to be more comprehensible.

Reply:
Great point, I'll incorporate this feedback throughout the page